ui/src/views/Quickstart/index.jsx

```diff
@@ -207,32 +205,68 @@ export default class QuickStart extends Component {
     this.setState(initialState);
   };
 
+  getMatchCondition = events => {
+    let condition = '';
+    const eventsJoin = Array.from(events).join(' ');
+
+    if (eventsJoin.includes('pull_request')) {
+      condition = `${condition}(tasks_for == "github-pull-request" && event["action"] in [${[
+        ...events,
+      ].sort()}])`;
+    }
+
+    if (eventsJoin.includes('push')) {
+      if (condition.length > 0) {
+        condition = `${condition} || `;
+      }
+
+      condition = `${condition}(tasks_for == "github-push")`;
+    }
+
+    if (eventsJoin.includes('release')) {
+      if (condition.length > 0) {
+        condition = `${condition} || `;
+      }
+
+      condition = `${condition}(tasks_for == "github-relase")`;
+    }
+
+    return condition;
+  };
+
   renderEditor() {
+    const {
+      access,
+      commands,
+      events,
+      image,
+      taskName,
+      taskDescription,
+    } = this.state;
     const newYaml = safeDump({
       ...initialYaml,
-      access: this.state.access,
-      tasks: [
-        {
-          ...initialYaml.tasks[0],
-          ...{
-            metadata: {
-              ...initialYaml.tasks[0].metadata,
-              name: this.state.taskName,
-              description: this.state.taskDescription,
-            },
-            extra: {
-              github: {
-                events: [...this.state.events].sort(),
+      policy: {
+        pullRequests: access,
+      },
+      tasks: {
+        $match: {
+          [`${this.getMatchCondition(events)}`]: {
+            ...initialYaml.tasks.$match,
+            ...{
+              metadata: {
+                ...initialYaml.tasks.$match.metadata,
+                name: taskName,
+                description: taskDescription,
+              },
+              payload: {
+                ...initialYaml.tasks.$match.payload,
+                image,
+                command: commands,
               },
-            },
-            payload: {
-              ...initialYaml.tasks[0].payload,
-              command: this.state.commands,
-              image: this.state.image,
             },
           },
         },
-      ],
+      },
     });
 
     return (
```

Comment:
Clicking the "Reset Form & File" button doesn't seem to reset the task definition editor. Would you be able to make it work, please? Happy to help if you have any questions.

<img width="85" alt="screen shot 2019-02-19 at 2 21 04 pm" src="https://user-images.githubusercontent.com/3766511/53041352-c09caf00-3451-11e9-897f-cbc4ba49a433.png">

_Note: It was probably never working :( so I take the full blame._

Reply:
Done